src/libsystemd/sd-device/device-monitor.c

```diff
@@ -450,6 +459,25 @@ static int passes_filter(sd_device_monitor *m, sd_device *device) {
         return device_match_parent(device, m->match_parent_filter, m->nomatch_parent_filter);
 }
 
+static bool check_sender_uid(sd_device_monitor *m, uid_t uid, bool is_unicast) {
+        assert(m);
+
+        /* Always trust messages from uid 0. */
+        if (uid == 0)
+                return true;
+
+        /* Trust messages come from outside of the current user namespace. */
+        if (!uid_range_contains(m->uid_range, m->n_uid_range, uid))
```

Comment:
please load the uid ranges lazily here and move this to the end, since it's the least likely to be needed and relatively expensive to load the uid range object